porch/pkg/registry/porch/package.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package porch
```

Comment:
Most of the contents of this file is boilerplate copied from the package revision implementation

Reply:
Is there a way we can split some of this out into a common library? We have upgraded to go 1.18, so we can use generics.

Reply:
Done! I combined it into one common library. Ended up not needing generics to do so. The new combined boilerplate is in the file `porch/pkg/registry/porch/packageobjects.go`.